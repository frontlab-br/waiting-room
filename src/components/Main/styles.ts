import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #f4f4f4;
  color: #000;
  width: 100%;
  height: 100%;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 4rem;
`

export const Title = styled.h1`
  font-size: 4rem;
  margin-bottom: 1rem;
`

export const Subtitle = styled.h2`
  font-size: 1.5rem;
  font-weight: 400;
  margin-bottom: 6rem;
`

export const Illustration = styled.img`
  width: min(30rem, 100%);
  margin-bottom: 6rem;
`
export const Logo = styled.img`
  width: 10rem;
`
