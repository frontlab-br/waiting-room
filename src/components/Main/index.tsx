import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Title>React Boilerplate</S.Title>
    <S.Subtitle>SSR . Precommit . Typescript . Tests</S.Subtitle>
    <S.Illustration src="AwetiMask.svg" alt="Aweti Mask" />
    <S.Logo src="/vercel.svg" alt="Vercel" />
  </S.Wrapper>
)

export default Main
