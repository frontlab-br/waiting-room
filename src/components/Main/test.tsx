import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('shold render the heading', () => {
    const { container } = render(<Main />)

    expect(screen.getByRole('heading', { name: /react/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('shold render the color correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({
      'background-color': '#f4f4f4'
    })
  })
})
