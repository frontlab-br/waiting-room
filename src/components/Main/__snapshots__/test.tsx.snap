// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> shold render the heading 1`] = `
.c0 {
  background-color: #f4f4f4;
  color: #000;
  width: 100%;
  height: 100%;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 4rem;
}

.c1 {
  font-size: 4rem;
  margin-bottom: 1rem;
}

.c2 {
  font-size: 1.5rem;
  font-weight: 400;
  margin-bottom: 6rem;
}

.c3 {
  width: min(30rem,100%);
  margin-bottom: 6rem;
}

.c4 {
  width: 10rem;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    React Boilerplate
  </h1>
  <h2
    class="c2"
  >
    SSR . Precommit . Typescript . Tests
  </h2>
  <img
    alt="Aweti Mask"
    class="c3"
    src="AwetiMask.svg"
  />
  <img
    alt="Vercel"
    class="c4"
    src="/vercel.svg"
  />
</main>
`
